R-packages/evalcast/R/signal_listcols.R

```diff
@@ -0,0 +1,32 @@
+signal_listcols <- function(signals, forecast_date) {
+  
+  cn <- colnames(signals)
+  ns <- nrow(signals)
+  # start_day should be NULL or date. We handle date/chr/fun(forecast_date) input
+  if ("start_day" %in% cn && is.list(signals$start_day)) {
+    std <- Date(ns)
+    for (i in 1:ns) {
+      if (is.function(signals$start_day[[i]])) {
+        std[i] <- as.Date(signals$start_day[[i]](forecast_date))
+      }
+    }
+    signals$start_day <- std
+  }
+  
+  if (! "as_of" %in% cn) signals$as_of <- forecast_date
```

Comment:
would be nice to combine or simplify this in some way:
```
if ("as_of" %in% cn) {
  if (is.list(signals$as_of)) {
    ...
  }
} else {
  signals$as_of <- forecast_date
}
```
or 
```
if (! "as_of" %in% cn) ...
if (is.list(signals)) {
  ...
}
```
or 
```
if (! "as_of" %in% cn) {
  ...
} else if (is.list(signals)) {
  ...
}
```

Reply:
Done.